backend/gn_module_monitoring/command/utils.py

```diff
@@ -116,6 +116,137 @@ def process_available_permissions(module_code, session):
+            ON p.id_role = ep.id_role
+            AND  p.id_action = ep.id_action
+            AND  p.id_module = ep.id_module
+            AND  p.id_object = ap.id_object
+            WHERE p.id_permission IS NULL
+        )
+        INSERT INTO gn_permissions.t_permissions
+        (id_role, id_action, id_module, id_object, scope_value, sensitivity_filter)
+        SELECT id_role, id_action, id_module, id_object, scope_value, sensitivity_filter
+        FROM new_p;
+    """
+        ).bindparams(module_code=module_code)
+    )
+
+    #  Suppression des permissions available inutile
+    #  on conserve POUR all
+    #  R : accès au module
+    #  U : modification des paramètres du module
+    #  E : Exporter les données du module
+    print(" Suppression des permissions available inutile")
+    session.execute(
+        text(
+            """
+        WITH to_del AS (
+            SELECT tp.*
+            FROM gn_permissions.t_permissions_available AS tp
+            JOIN gn_commons.t_modules AS tm
+            ON tm.id_module = tp.id_module AND tm."type" = :module_code
+            JOIN gn_permissions.t_objects AS o
+            ON o.id_object = tp.id_object AND code_object = 'ALL'
+            JOIN gn_permissions.bib_actions AS ba
+            ON tp.id_action = ba.id_action AND NOT ba.code_action  IN ('R', 'E', 'U')
+        )
+        DELETE FROM gn_permissions.t_permissions_available AS tp
+        USING to_del td
+        WHERE  tp.id_module = td.id_module
+        AND tp.id_object = td.id_object
+        AND tp.id_action = td.id_action
+        AND tp."label" = td."label"
+        AND tp.scope_filter = td.scope_filter
+        AND tp.sensitivity_filter = td.sensitivity_filter;
+    """
+        ).bindparams(module_code=module_code)
+    )
+    #  Suppression des objects ALL
+    print(" Suppression des objects ALL")
+    session.execute(
+        text(
+            """
+            WITH to_del AS (
+                SELECT tp.*
+                FROM gn_permissions.t_permissions_available AS tp
+                JOIN gn_commons.t_modules AS tm
+                ON tm.id_module = tp.id_module AND tm.module_code = :module_code
+                JOIN gn_permissions.t_objects AS o
+                ON o.id_object = tp.id_object AND code_object = 'ALL'
+            )
+            DELETE FROM gn_permissions.t_permissions_available AS tp
+            USING to_del td
+            WHERE  tp.id_module = td.id_module
+            AND tp.id_object = td.id_object
+            AND tp.id_action = td.id_action
+            AND tp."label" = td."label"
+            AND tp.scope_filter = td.scope_filter
+            AND tp.sensitivity_filter = td.sensitivity_filter;
+            """
+        ).bindparams(module_code=module_code)
+    )
+
+    # Suppression des permissions qui ne sont pas dans les permissions available
+    print(" Suppression des permissions qui ne sont pas dans les permissions available")
+    session.execute(
```

Comment:
**Suggestion:** Consider using logging instead of print statements for better control over output and to include additional information such as timestamp, file, line number etc. [best practice]
```suggestion
    import logging
    logger = logging.getLogger(__name__)

    logger.info(f" Reprise des permissions du module {module_code}")
    logger.info(" Création des permissions des sous-objets à partir des permissions contenus dans l'objet ALL")
    logger.info(" Suppression des permissions available inutile")
    logger.info(" Suppression des objects ALL")
    logger.info(" Suppression des permissions qui ne sont pas dans les permissions available")
```